tooling/artifact_cli/src/bin/execute.rs

```diff
@@ -1,33 +1,91 @@
 #![forbid(unsafe_code)]
 
+use std::path::{Path, PathBuf};
+
+use acvm::compiler::validator::validate_witness;
+use bn254_blackbox_solver::Bn254BlackBoxSolver;
 use clap::{Parser, Subcommand, command};
 use color_eyre::eyre;
 use const_format::formatcp;
 use tracing_subscriber::{EnvFilter, fmt::format::FmtSpan};
 
-use noir_artifact_cli::commands::execute_cmd;
+use noir_artifact_cli::{Artifact, commands::execute_cmd, fs::witness::load_witness_from_file};
+use noirc_driver::CompiledProgram;
 
 const PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 static VERSION_STRING: &str = formatcp!("version = {}\n", PKG_VERSION,);
 
+fn parse_and_normalize_path(path: &str) -> Result<PathBuf, String> {
+    let path = PathBuf::from(path);
+    Ok(path)
+}
+
 #[derive(Parser, Debug)]
 #[command(name="noir-execute", author, version=VERSION_STRING, about, long_about = None)]
 struct ExecutorCli {
-    #[command(flatten)]
-    command: execute_cmd::ExecuteCommand,
+    #[command(subcommand)]
+    command: Command,
 }
 
-#[non_exhaustive]
-#[derive(Subcommand, Clone, Debug)]
-enum ArtifactCommand {
+#[derive(Subcommand, Debug)]
+enum Command {
+    /// Execute a binary program or a circuit artifact.
     Execute(execute_cmd::ExecuteCommand),
```

Comment:
It looks like with this the existing usage changes from `noir-execute <args>` to `noir-execute execute <args>`. This is why I wanted to rename to `noir-executor execute, which @TomAFrench didn't like, and why I thought we could have a separate `noir-check-witness`, and maybe deal with the duplication of the general CLI boilerplate in a different way.

There is a trick [here](https://stackoverflow.com/a/79564853) to make `noir-execute <args>` working. 

Otherwise I think you have to update `examples/oracle_transcript/log_and_exec_transcript.sh`

Reply:
Are you sure?
```console
$ cargo run -q -p noir_artifact_cli -- execute --artifact-path target/oracle_transcript.json  --oracle-file Oracle.jsonl  --prover-file Prover.toml  --output-dir target

No such file or directory (os error 2)

$ cargo run -q -p noir_artifact_cli --  --artifact-path target/oracle_transcript.json  --oracle-file Oracle.jsonl  --prover-file Prover.toml  --output-dir target

error: unexpected argument '--artifact-path' found

Usage: noir-execute <COMMAND>

For more information, try '--help'.
```